src/components/AgronodInputs/AgronodTextField/AgronodTextField.tsx

```diff
@@ -0,0 +1,205 @@
+  endAndorment,
+  ...rest
+}: AgronodTextFieldProps) => {
+  const icon = useMemo(() => {
+    if (status === "error" && hasIcon) {
+      return <AgronodIcon name="errorContained" color="error" />;
+    }
+    if (status === "warning" && hasIcon) {
+      return <AgronodIcon name="warningContained" color="warning" />;
+    }
+    return null;
+  }, [status]);
+
+  return (
+    <MuiTextField
+      InputLabelProps={{
+        color: "secondary",
+        shrink: false,
+      }}
+      InputProps={{
+        startAdornment: icon !== null && (
+          <InputAdornment position="start">{icon}</InputAdornment>
+        ),
+        endAdornment: endAndorment !== undefined && (
+          <InputAdornment position="start">{endAndorment}</InputAdornment>
+        ),
+      }}
+      variant="outlined"
+      sx={(theme) => ({
+        "& .MuiOutlinedInput-root:not(.Mui-disabled)": {
+          backgroundColor:
+            status === "warning"
+              ? theme.palette.warning.pastel
+              : status === "error"
+                ? theme.palette.error.pastel
+                : undefined,
+
+          "& fieldset": {
+            borderColor:
+              status === "warning" ? theme.palette.warning.main : undefined,
+          },
+
+          "&:hover fieldset": {
+            borderColor:
+              status === "warning" ? theme.palette.warning.main : undefined,
+          },
+
+          "&.Mui-focused fieldset": {
+            borderColor:
+              status === "warning" ? theme.palette.warning.main : undefined,
+          },
+        },
+        "& .MuiOutlinedInput-input": {
+          textAlign: textAligment ? textAligment : "left",
+        },
+        "& .MuiOutlinedInput-root:has(input[value='']):not(:has(.MuiChip-root)):not(.Mui-disabled)":
+          {
+            backgroundColor:
+              emptyStyle === "highlighted"
+                ? theme.palette.secondary.pastel
+                : undefined,
+          },
+      })}
+      error={status === "error"}
+      helperText={
+        !hideHelperText && helperText !== undefined ? helperText : undefined
+      }
+      {...rest}
+    />
+  );
+};
+
+const AgronodTextField = ({
+  status = "default",
+  emptyStyle = "default",
+  helperText,
+  hideHelperText,
+  tooltipText,
+  hasIcon = false,
+  label,
+  textAligment = "left",
+  endAndorment,
+  ...rest
+}: AgronodTextFieldProps) => {
+  const [showTooltip, setShowTooltip] = useState(false);
+
+  const generateTooltipText = useMemo(() => {
+    if ((status === "error" || status === "warning") && tooltipText) {
+      return tooltipText;
+    } else {
+      return tooltipText;
+    }
+  }, [status, tooltipText]);
+
+  return (
+    <Box
+      onMouseEnter={() => setShowTooltip(true)}
+      onMouseLeave={() => setShowTooltip(false)}
+    >
+      <Tooltip
+        open={showTooltip && !!generateTooltipText}
+        placement="top"
+        arrow
+        title={generateTooltipText ? generateTooltipText : ""}
```

Comment:
use tooltipText instead of this memo function since it always return tooltipText